tests/unit/db_scripts/test_garbage_collector.py

```diff
@@ -169,75 +200,544 @@ def test_prompt_for_confirmation_invalid_then_valid(self, mocker: MockerFixture,
+
+        Case 1: is_accessible_mount=True but workspace_exists=False
+        Expected: Workspace flagged as invalid.
+
+        Args:
+            mocker: Pytest mocker fixture.
+            gc: DatabaseGarbageCollector instance.
+            mock_run: Mocked run instance.
+            mock_db: Mocked database instance.
+        """
+        workspace = "/mnt/shared/missing_workspace"
+        mock_run.get_workspace.return_value = workspace
+        mock_run.get_id.return_value = "run-123"
+        mock_db.runs.get_all.return_value = [mock_run]
+
+        mocker.patch.object(gc, "_is_workspace_on_accessible_mount", return_value=True)
+        mocker.patch("os.path.exists", return_value=False)
+
+        gc.check_run_workspaces()
+
+        assert len(gc._issues["run"]) == 1
+        assert gc._issues["run"][0] == mock_run
+
+        assert len(gc._issues["inaccessible_runs"]) == 0
+
+    def test_check_with_workspace_on_inaccessible_mount(
+        self, mocker: MockerFixture, gc: DatabaseGarbageCollector, mock_run: MagicMock, mock_db: MagicMock
+    ):
+        """
+        Test checking runs when workspace is on an inaccessible mount.
+
+        Case 2: is_accessible_mount=False and workspace_exists=False
+        Expected: Counted as inaccessible, not flagged as invalid.
+
+        Args:
+            mocker: Pytest mocker fixture.
+            gc: DatabaseGarbageCollector instance.
+            mock_run: Mocked run instance.
+            mock_db: Mocked database instance.
+        """
+        workspace = "/p/lustre3/workspace"
+        mock_run.get_workspace.return_value = workspace
+        mock_run.get_id.return_value = "run-123"
+        mock_db.runs.get_all.return_value = [mock_run]
+
+        mocker.patch.object(gc, "_is_workspace_on_accessible_mount", return_value=False)
+        mocker.patch("os.path.exists", return_value=False)
+
+        mock_log = mocker.patch("merlin.db_scripts.garbage_collector.LOG")
+
+        gc.check_run_workspaces()
+
+        # Should NOT be in issues list for run entries
+        assert len(gc._issues["run"]) == 0
+
+        # Should be in issues list for inaccessible runs
+        assert len(gc._issues["inaccessible_runs"]) == 1
+        assert gc._issues["inaccessible_runs"][0] == mock_run
+
+        # Should log warning about inaccessible workspaces
+        mock_log.warning.assert_called()
+        warning_calls = [str(call) for call in list(mock_log.warning.call_args_list + mock_log.debug.call_args_list)]
+        assert any("inaccessible" in str(call).lower() for call in warning_calls)
+
+    def test_check_with_valid_workspace_on_root_filesystem(
         self, mocker: MockerFixture, gc: DatabaseGarbageCollector, mock_run: MagicMock, mock_db: MagicMock
     ):
         """
-        Test checking runs when workspace exists.
+        Test checking runs when workspace exists on root filesystem (e.g., /tmp).
+
+        Case 3: is_accessible_mount=False but workspace_exists=True
```

Comment:
shouldn't this be 'is_accessible_mount=True' since /tmp will exist?

Reply:
I think this is a naming convention issue. The answer to your question is technically yes. However, the `check_run_workspaces()` method of the `DatabaseGarbageCollector` (which is what's being tested here) ends up calling `get_accessible_mounts()` with `exclude_root=True`. Since this `exclude_root=True` option is enabled, the "/tmp" directory will not be recognized as an accessible mount; instead it acts as a path on the root filesystem. Therefore, in this test we simulate this by mocking the return value of `_is_workspace_on_accessible_mount()` to be `False`.

Maybe I should change the workspace path here to just be "/local_workspace"?